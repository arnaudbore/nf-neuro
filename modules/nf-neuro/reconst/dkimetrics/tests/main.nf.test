nextflow_process {

    name "Test Process RECONST_DKIMETRICS"
    script "../main.nf"
    process "RECONST_DKIMETRICS"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "reconst"
    tag "reconst/dkimetrics"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "DWIms1000-2000-dir10-10.zip", "segmentation.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("reconst - dkimetrics") {
        when {
            process {
                """
                ch_split_test_data = LOAD_DATA.out.test_data_directory
                    .branch{
                        dwi: it.simpleName == "DWIms1000-2000-dir10-10"
                        segmentation: it.simpleName == "segmentation"
                    }
                ch_dwi = ch_split_test_data.dwi.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/dwi.nii.gz"),
                        file("\${test_data_directory}/dwi.bval"),
                        file("\${test_data_directory}/dwi.bvec")
                    ]
                }
                ch_mask = ch_split_test_data.segmentation.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/brainmask/brainmask.nii.gz")
                    ]
                }
                input[0] = ch_dwi
                    .join(ch_mask)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    niftiMD5SUM(process.out.ak.get(0).get(1), 6),
                    niftiMD5SUM(process.out.dki_ad.get(0).get(1), 6),
                    niftiMD5SUM(process.out.dki_fa.get(0).get(1), 6),
                    niftiMD5SUM(process.out.dki_md.get(0).get(1), 6),
                    niftiMD5SUM(process.out.dki_rd.get(0).get(1), 6),
                    niftiMD5SUM(process.out.dki_residual.get(0).get(1), 6),
                    niftiMD5SUM(process.out.mk.get(0).get(1), 6),
                    niftiMD5SUM(process.out.msd.get(0).get(1), 6),
                    niftiMD5SUM(process.out.rk.get(0).get(1), 6),
                    file(process.out.mqc.get(0).get(1)).name,
                    process.out.versions
                ).match() }
            )
        }
    }

    test("reconst - dkimetrics_light") {
        config "./nextflow_light.config"
        when {
            process {
                """
                ch_split_test_data = LOAD_DATA.out.test_data_directory
                    .branch{
                        dwi: it.simpleName == "DWIms1000-2000-dir10-10"
                        segmentation: it.simpleName == "segmentation"
                    }
                ch_dwi = ch_split_test_data.dwi.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/dwi.nii.gz"),
                        file("\${test_data_directory}/dwi.bval"),
                        file("\${test_data_directory}/dwi.bvec")
                    ]
                }
                ch_mask = ch_split_test_data.segmentation.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/brainmask/brainmask.nii.gz")
                    ]
                }
                input[0] = ch_dwi
                    .join(ch_mask)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    niftiMD5SUM(process.out.dki_fa.get(0).get(1), 6),
                    niftiMD5SUM(process.out.dki_ad.get(0).get(1), 6),
                    niftiMD5SUM(process.out.dki_md.get(0).get(1), 6),
                    niftiMD5SUM(process.out.dki_rd.get(0).get(1), 6),
                    process.out.versions
                ).match() }
            )
        }
    }

    test("reconst - stub-run dkimetrics") {
        tag "stub"
        options "-stub-run"
        when {
            process {
                """
                ch_split_test_data = LOAD_DATA.out.test_data_directory
                    .branch{
                        dwi: it.simpleName == "DWIss1000-dir32"
                        segmentation: it.simpleName == "segmentation"
                    }
                input[0] = ch_split_test_data.dwi.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/dwi.nii.gz"),
                        file("\${test_data_directory}/dwi.bval"),
                        file("\${test_data_directory}/dwi.bvec"),
                        []
                    ]
                }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() }
            )
        }
    }
}
