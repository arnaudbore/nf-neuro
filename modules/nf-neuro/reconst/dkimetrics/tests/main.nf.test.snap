{
    "reconst - stub-run dkimetrics": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-15T17:52:47.854638003"
    },
    "reconst - dkimetrics_light": {
        "content": [
            "test__dki_fa.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,bd4d9ba08cf1d6d38c271950b5003d1c",
            "test__dki_ad.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,59c64827ef5ed43c5e37632fcbbc3312",
            "test__dki_md.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,f45a78a08fe054fc59a47b00bec528e5",
            "test__dki_rd.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,73578541bf1219f26c93a68b53c93dbb",
            [
                "versions.yml:md5,254c1fb19cea238e381632d5ac4d93ba"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-15T17:52:19.051125326"
    },
    "reconst - dkimetrics": {
        "content": [
            "test__ak.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,323c7e06b2f0d4b4949964d3fbaa68f8",
            "test__dki_ad.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,59c64827ef5ed43c5e37632fcbbc3312",
            "test__dki_fa.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,bd4d9ba08cf1d6d38c271950b5003d1c",
            "test__dki_md.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,f45a78a08fe054fc59a47b00bec528e5",
            "test__dki_rd.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,73578541bf1219f26c93a68b53c93dbb",
            "test__dki_residual.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,cd704db10e562429bb817d40d72de661",
            "test__mk.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,3fa463c05280f3c9f6bfdfb38ba868a8",
            "test__msd.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,1dbd63dc9a4fd2b18eba652499fe2131",
            "test__rk.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,fb250d550753fb1c0bbdc304e30739b5",
            "test__dki_mqc.png",
            [
                "versions.yml:md5,254c1fb19cea238e381632d5ac4d93ba"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-15T17:42:38.429429555"
    }
}